src/database.ts

```diff
@@ -0,0 +1,88 @@
+import { initializeApp, applicationDefault, cert } from 'firebase-admin/app'
+import serviceAccountCredentials from '/home/jchae/projects/findadoc-server/find-a-doc-japan-firebase-adminsdk-k7f1f-12c410ff49.json'
+import { getFirestore, Timestamp, FieldValue } from 'firebase-admin/firestore'
+
+export async function initializeDb() {
+  initializeApp({
+    credential: cert(serviceAccountCredentials)
+  })
+}
+
+export async function addHealthcareProfessional(healthcareProfessionalsRef, healthcareProfessional) {
+  healthcareProfessionalsRef.add(healthcareProfessional)
+}
+
+export async function addFacility(facilityRef, facility) {
+  facilityRef.add(transformFacilityForFirestore(facility))
+}
+
+function transformFacilityForFirestore(facility) {
+  const healthcareProfessionalIds = facility.healthcareProfessionals.map(hp => hp.id)
+  facility.healthcareProfessionals = healthcareProfessionalIds
+
+  return facility
+}
+
+export async function getHealthcareProfessionals() {
+    const db = getFirestore()
+    const hpRef = db.collection('healthcareProfessionals')
+
+    const snapshot = await hpRef.get()
+  var healthcareProfessionals = []
+    snapshot.forEach(doc => {
+      healthcareProfessionals.push(doc.data())
+    })
+
+  return healthcareProfessionals
+}
+
+export async function getFacilities() {
+    const db = getFirestore()
+    const facilitiesRef = db.collection('facilities')
+
+    const snapshot = await facilitiesRef.get()
+    var facilities = []
+    snapshot.forEach(doc => {
+      facilities.push(doc.data())
+    })
+
+    facilities = Promise.all(facilities.map(facility => hydrateFacility(facility))).then((resource) => resource)
```

Comment:
Nice catch! This is the proper wait o await a foreach loop. BUT! They just released a new syntax to solve this ✨

```suggestion
    for await (const facility of facilities) {
    await yourcode ()
  }
```